Source/Core/DolphinQt/HotkeyScheduler.cpp

```diff
@@ -139,10 +139,22 @@ void HotkeyScheduler::Run()
 
   while (!m_stop_requested.IsSet())
   {
-    Common::SleepCurrentThread(1000 / 60);
+    const u64 freelook_update_freq = 200;
+    const u64 interval = SystemTimers::GetTicksPerSecond() / freelook_update_freq;
+    const u64 now = CoreTiming::GetTicks();
 
-    if (Core::GetState() == Core::State::Uninitialized || Core::GetState() == Core::State::Paused)
+    if (now - m_last_freelook_ticked > interval)
+    {
+      m_last_freelook_ticked = now;
+      g_controller_interface.SetCurrentInputChannel(ciface::InputChannel::FreeLook);
       g_controller_interface.UpdateInput();
+      FreeLook::UpdateInput();
+    }
+
+    Common::SleepCurrentThread(1000 / 60);
```

Comment:
If this sleeps the thread for ~16ms, how is freelook going to update at 200Hz? `if (now - m_last_freelook_ticked > interval)` is always gonna be true.

Reply:
Why is the refresh rate 200Hz? Maybe it should be hinted. Is it because you want the input polling to be as up to date as possible when a new game frame starts? Or can freelook actually output more frames that the game naturally would (I don't know how it works)?